### PR TITLE
Improved Hide/Unhide Dataframe(s) dialog

### LINKED ISSUE #7533: Unhide Data Frame seems to not work
I have tried this by loading `dodoma` and `ghana` datasets but you can try with any datasets, then I have hidden `ghana` by clicking on Hide in dataview sheet menu context, it is fine  then when I tried to `Unhide` using `Hide/Unhide Data Frame` dialogue, it seems to not work.
The code generating by the `Hide/Unhide Data Frame` dialogue is 
image
But it works when I change this code to
image
@shadrackkibet @lilyclements what do you think? I have check with recent version of R-Instat this seems to not work too.

### PR DESCRIPTION
**Pull request by Vitalis95**
Fixes #7533 
@rdstern @N-thony @lloyddewit , I have added 2 radio buttons(Hide &Unhide), this improvements simplify the dialog for the user

### CODE REVIEW

**Review by rdstern — COMMENTED** [review]
@Vitalis95 I am not sure how this is supposed to work?
image

I had a number of data frames and hid some using the bottom right-click. (They just hid without showing the dialogue.) Then I pressed Unhide and it gave me your new dialogue.  I then tried to unhide those that were hidden, though it listed all of them.  And I couldn't use Add or double-click to put them into the receiver in unhide.  They did go into the receiver with Hide.  (But they were already hidden!

**Comment by Vitalis95** [comment]
@rdstern , have a look at it.

**Comment by N-thony** [comment]
@derekagorhom can you peer review this?

**Review by derekagorhom — DISMISSED** [review]

File: instat/dlgHideDataframes.vb [hunk 1]
```diff
@@ -43,23 +48,71 @@ Public Class dlgHideDataframes
...
     Private Sub TestOKEnabled()
-        ' You cannot hide all data frames. When the receiver is blank all data frames are unhidden so this is allowed.
-        If ucrReceiverMultiple.lstSelectedVariables.Items.Count <> ucrSelectorForDataFrames.lstAvailableVariable.Items.Count Then
-            ucrBase.OKEnabled(True)
+        If rdoUnhideDataFrame.Checked Then
+            If Not ucrReceiverMultipleUnhide.IsEmpty Then
+                ucrBase.OKEnabled(True)
+            Else
+                ucrBase.OKEnabled(False)
+            End If
```
Comment (lloyddewit) [inline instat/dlgHideDataframes.vb:106]: is this simpler?
```suggestion
            ucrBase.OKEnabled(ucrReceiverMultipleUnhide.IsEmpty)
```

File: instat/dlgHideDataframes.vb [hunk 2]
```diff
@@ -43,23 +48,71 @@ Public Class dlgHideDataframes
...
         Else
-            ucrBase.OKEnabled(False)
+            ' You cannot hide all data frames. When the receiver is blank all data frames are unhidden so this is allowed.
+            If ucrReceiverMultiple.lstSelectedVariables.Items.Count <> ucrSelectorForDataFrames.lstAvailableVariable.Items.Count Then
+                ucrBase.OKEnabled(True)
+            Else
+                ucrBase.OKEnabled(False)
+            End If
```
Comment (lloyddewit) [inline instat/dlgHideDataframes.vb:113]: see comment above

File: instat/dlgHideDataframes.vb [hunk 3]
```diff
@@ -69,7 +122,9 @@ Public Class dlgHideDataframes
     End Sub
 
     Public Sub SetRCodeForControls(bReset As Boolean)
-        SetRCode(Me, ucrBase.clsRsyntax.clsBaseFunction, bReset)
+
```
Comment (lloyddewit) [inline instat/dlgHideDataframes.vb:125]:
```suggestion
```

**Comment by N-thony** [comment]
@Vitalis95 is this ready for re-review?

**Comment by Vitalis95** [comment]
@N-thony , yes. Could you test it?

**Comment by Vitalis95** [comment]
@N-thony , the reset issue is now ok.

**Review by N-thony — APPROVED** [review]
@lloyddewit over to you

**Comment by rdstern** [comment]
@Vitalis95 this is looking pretty good now.
So Hide, clicking at the bottom just hides the current data frame, and Unhide takes you to the dialogue.
a) I was initially thinking that we should disable unhide, at the bottom, when there is nothing to hide.  But I now suggest we leave it as it is, as it is a neat way of getting to the Hide/Unhide dialogue!
b) But it is odd that when you click on Unhide you get to Hide in the dialogue.  Could you either swap the top radio button s round or go to the unhide (second one) initially.  I quite like Hide  before Unhide in the dialogue.  So I'd prefer to go to the second - Unhide - when you first click on Unhide at the bottom.  (Ideally you then have hide as the default if you first go to the dialogue itself.)  So keep it as it is now, if that's confusing!
c) But one change you need is that the dialogue remembers the old setting - and this time it shouldn't.  Because it is now unhidden!  So clear the Unhide receiver when you unhide.  And I assume you want to clear the Hide receiver also when you hide through the dialogue!!!

**Comment by Vitalis95** [comment]
@rdstern , have a look at it for part c).

**Review by rdstern — APPROVED** [review]
@Vitalis95 looks fine now.  Thanks.
Once @lloyddewit approves it will be good to merge.

**Review by lloyddewit — APPROVED** [review]

**Pull request merged**